lc2fen/split_board.py

```diff
@@ -62,25 +70,43 @@ def split_square_board_image(board_image, output_name, out_dir, board=None):
             )
 
 
-def split_board_image(board_image, square_corners, output_name, out_dir, board=None):
-    """
-    Splits a board into the individual squares using their corners.
-
-    Splits the board_image into the content of individual squares and
-    saves them into out_dir as output_name appended with the
-    corresponding position.
-
-    :param board_image: The board image to split.
-    :param square_corners: List with the coordinates of each of the
-        corners of the squares of the chessboard.
-    :param output_name: Starting name of the pieces.
-    :param out_dir: Output directory where to save the pieces.
-    :param board: If board is not None, it represents the 8x8 board
-        matrix with the corresponding labels in each position.
+def split_board_image_advanced(
+    board_image: str,
+    square_corners: list[tuple[int, int]],
+    output_name: str,
+    out_dir: str,
+    board: list[list[str]] = None,
+):
+    """Split a chessboard image into 64 square images using the square-corner coordinates.
+
+    This function splits a chessboard image into 64 images of the 64 squares using
```

Comment:
Could you add a comment saying that this is not currently used in the main flow of the framework although it would be a good (but complicated) improvement? This is because the CNNs were trained with a dataset that cropped the images using the trivial method instead of this more advanced one. If the CNNs were trained using a dataset cropped with this advanced method, the overall accuracy would improve (I detail this in section 2.2.1 of my bachelor thesis (unfortunately only in Spanish [here](https://docta.ucm.es/bitstreams/eaca2ef1-b72f-49aa-9eb4-dd97488c0773/download), but the image can give an idea).

Reply:
Very interesting, and I just added the comment!